src/providers/google/gdrive-document.ts

```diff
@@ -0,0 +1,207 @@
+            syncPosition = this.stopSync(syncPosition);
+
+            return {
+                position: syncPosition,
+                results: [],
+            };
+        }
+
+        const results = await this.buildResults(
+            drive,
+            auth,
+            serverResponse,
+            syncPosition.breakId,
+            _.has(this.config, "metadata.breakTimestamp")
+                ? this.config.metadata.breakTimestamp
+                : undefined
+        );
+
+        syncPosition = this.setNextPosition(syncPosition, serverResponse);
+
+        if (results.length != this.config.batchSize) {
+            // Not a full page of results, so stop sync
+            syncPosition.syncMessage = `Processed ${results.length} items. Stopping. No more results.`;
+            syncPosition = this.stopSync(syncPosition);
+        }
+
+        return {
+            results,
+            position: syncPosition,
+        };
+    }
+
+    protected stopSync(syncPosition: SyncHandlerPosition): SyncHandlerPosition {
+        if (syncPosition.status == SyncHandlerStatus.ENABLED) {
+            return syncPosition;
+        }
+
+        syncPosition.status = SyncHandlerStatus.ENABLED;
+        syncPosition.thisRef = undefined;
+        syncPosition.breakId = syncPosition.futureBreakId;
+        syncPosition.futureBreakId = undefined;
+
+        return syncPosition;
+    }
+
+    protected setNextPosition(
+        syncPosition: SyncHandlerPosition,
+        serverResponse: GaxiosResponse<drive_v3.Schema$FileList>
+    ): SyncHandlerPosition {
+        if (!syncPosition.futureBreakId && serverResponse.data.files.length) {
+            syncPosition.futureBreakId = serverResponse.data.files[0].id;
+        }
+
+        if (_.has(serverResponse, "data.nextPageToken")) {
+            // Have more results, so set the next page ready for the next request
+            syncPosition.syncMessage = `Batch complete (${this.config.batchSize}). More results pending.`;
+            syncPosition.thisRef = serverResponse.data.nextPageToken;
+        } else {
+            syncPosition.syncMessage = "Stopping. No more results.";
+            syncPosition = this.stopSync(syncPosition);
+        }
+
+        return syncPosition;
+    }
+
+    protected async buildResults(
+        drive: drive_v3.Drive,
+        auth: OAuth2Client,
+        serverResponse: GaxiosResponse<drive_v3.Schema$FileList>,
+        breakId: string,
+        breakTimestamp?: string
+    ): Promise<SchemaDocument[]> {
+        const results: SchemaDocument[] = [];
+        for (const file of serverResponse.data.files) {
+            const fileId = file.id;
+
+            if (fileId == breakId) {
+                const logEvent: SyncProviderLogEvent = {
+                    level: SyncProviderLogLevel.DEBUG,
+                    message: `Break ID hit (${breakId})`
+                }
+                this.emit('log', logEvent)
```

Comment:
Can't set as "Unknown" as it's not an ISO Date/time. Set it to now.

Reply:
> Can't set as "Unknown" as it's not an ISO Date/time. Set it to now.

Fixed